### ISSUE TITLE
Snyk Vuln Report: High Severity Prototype Pollution through `defaults-deep`

### ISSUE THREAD

**Issue by corpetty** [libp2p/js-libp2p-circuit#45]
When running snyk.io on our repo at `status-im/chat`, we're getting a vuln report linking up to here.  The details are as follows:

- Vulnerable module: `defaults-deep`
    - Introduced through: `ipfs@0.33.1`

#### Detailed paths and remediation
- Introduced through: `status-js-web@0.1.0` › `ipfs@0.33.1` › `libp2p@0.23.1` › `libp2p-switch@0.40.8` › `libp2p-circuit@0.2.1` › `defaults-deep@0.2.4`
    - Remediation: No remediation path available.

#### Overview
`defaults-deep` is a like extend but recursively copies only the missing properties/values to the target object.

Affected versions of this package are vulnerable to Prototype Pollution. Provided certain input defaults-deep can add or modify properties of the  Object prototype. These properties will be present on all objects.

#### PoC

```javascript
var defaultsDeep = require('defaults-deep');
var payload = JSON.parse('{"constructor": {"prototype": {"isAdmin": true}}}');
defaultsDeep({}, payload);
console.log({}.isAdmin); // true
```

#### More Info
[NPM Security Advisory](https://www.npmjs.com/advisories/778)
[HackerOne Report](https://hackerone.com/reports/380878)

**Comment by dryajov** [comment]
Interesting, thanks for bringing this up. I believe other modules in the stack might rely on it, I know that ipfsd-ctl used to rely on it (probably still is), but that is not as critical as it is more of a utility/testing tool.

**Comment by jacobheun** [comment]
We recently removed the package as it wasn't being used. I just checked version 0.2.1 and the package is included but not being used anywhere, so this shouldn't be an issue.

**Comment by jacobheun** [comment]
It was added in `0.2.1` but never used. I checked each version between 0.2.1 and 0.3.5 to verify it wasn't being used anywhere.

Here's the compare of the addition in 0.2.1: https://github.com/libp2p/js-libp2p-circuit/compare/v0.2.0...v0.2.1#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R57

**Comment by mcollina** [comment]
You should use https://www.npmjs.com/package/bourne to fix this.

**Comment by jacobheun** [comment]
I've published `libp2p-circuit@0.2.2` which removes the extraneous package. The latest `0.3` versions of `libp2p-circuit` do not contain the package.

As it wasn't being used no replacement was needed and there should be no risk. The package was likely being used during feature development and wasn't properly groomed before being merged.

Thanks for sharing the `bourne` package, if we need to do any replacing elsewhere we'll leverage that.
